### ISSUE TITLE
NSFW Exit redirects. Seems likely to share script.

### ISSUE THREAD

**Issue by spirillen** [easylist/easylist#9079]
### List the website(s) you're having issues:
The following table will contain all sites that is exit circulating to each other, as some of us absolutely prefer one issue for each domain; there will be linked to individual issues for that. 

- This particular thread is about, what I'll assume will be the same exit script.
- I will also endure other to find and submit the exit script used on these pages 

Link circus members

| Domain | External Url |
| :----- | :----------- |
| bestfreecams.club | https://mypdns.org/my-privacy-dns/porn-records/-/issues/616 |
| publicxxxvideo.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/619 |
| sessotube.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/505 |
| publicxxxvideo.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/619 |
| xxxporn.media | https://mypdns.org/my-privacy-dns/porn-records/-/issues/620 |
| japanxxx.asia | https://mypdns.org/my-privacy-dns/porn-records/-/issues/622 |
| ixxx.fit | https://mypdns.org/my-privacy-dns/porn-records/-/issues/462 |
| germanteenporno.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/624 |
| firstxnxx.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/625 |
| youngxxxtubes.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/626 |
| xnxxmilf.asia | https://mypdns.org/my-privacy-dns/porn-records/-/issues/627 |
| pearlytube.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/628 |
| privatesexfilms.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/629 |
| bigtitslove.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/631 |
| xxxvideo4.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/632 |
| ixxx.asia | https://mypdns.org/my-privacy-dns/porn-records/-/issues/633 |
| trafficoxtop.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/634 |
| sexmovies.su | https://mypdns.org/my-privacy-dns/porn-records/-/issues/635 |
| xxxadultsex.net | https://mypdns.org/my-privacy-dns/porn-records/-/issues/636 | 
| pichunter.vip | https://mypdns.org/my-privacy-dns/porn-records/-/issues/637 |
| pornotube89.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/623 |
| easternexxxpress.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/638 |
|barely-legal-porn.com (Auto redirected to `pornyoung.pro`)| https://mypdns.org/my-privacy-dns/porn-records/-/issues/639 |
| pornyoung.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/640 |
| blackporn.asia | https://mypdns.org/my-privacy-dns/porn-records/-/issues/641 |
| youjizzz.club | https://mypdns.org/my-privacy-dns/porn-records/-/issues/281 |
| freeporn.works | https://mypdns.org/my-privacy-dns/porn-records/-/issues/277 |
| privatesexfilms.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/629 |
| realityhomemade.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/642 |
| hometubeporn.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/643 |
| vintagetube.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/644 |
| beeg.ink | https://mypdns.org/my-privacy-dns/porn-records/-/issues/645 |
| txxx.asia | https://mypdns.org/my-privacy-dns/porn-records/-/issues/274 |
| xboul.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/508 |
| xhamster-art.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/444 |
| freexxx.best | https://mypdns.org/my-privacy-dns/porn-records/-/issues/646 |
| publicxxxvideo.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/269 |
| xxxvideodownload.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/648 |
| hotmilfclips.net | https://mypdns.org/my-privacy-dns/porn-records/-/issues/263 |
| x-videos.su | https://mypdns.org/my-privacy-dns/porn-records/-/issues/649 |
| hdsex.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/650 |
| xxnxx.fun | https://mypdns.org/my-privacy-dns/porn-records/-/issues/651 |
| hdsexvideos.pro | https://mypdns.org/my-privacy-dns/porn-records/-/issues/647 |
| pornxxx.run | https://mypdns.org/my-privacy-dns/porn-records/-/issues/652 |
| xxxmovies.world | https://mypdns.org/my-privacy-dns/porn-records/-/issues/402 |
|  redwap.wtf | https://mypdns.org/my-privacy-dns/porn-records/-/issues/131 |
| pussiporn.com | https://mypdns.org/my-privacy-dns/porn-records/-/issues/653 |
| tiava.asia | https://mypdns.org/my-privacy-dns/porn-records/-/issues/387 |
| empflix.icu | https://mypdns.org/my-privacy-dns/porn-records/-/issues/654 |
| porzo.club | https://mypdns.org/my-privacy-dns/porn-records/-/issues/655  |
| xxxclub.net | https://mypdns.org/my-privacy-dns/porn-records/-/issues/656 |

## External relations?
See above table

### What happens?

 - [x] Exit popup -under
 - [X] Exit script redirecting to other porno sites
 - [ ] AdWare 
 - [ ] Tracking

## Screenshot
Non applicable in this issue

### List Subscriptions you're using:

- uBlock filters
- uBlock filters – Badware risks
- uBlock filters – Privacy
- uBlock filters – Resource abuse
- EasyList
- EasyPrivacy
- Online Malicious URL Blocklist
- Peter Lowe’s Ad and tracking server list
- Spirillen's – Adblock Rules (https://mypdns.org/my-external-stuff/ublockorigin-rules/)

### Your settings

- OS/version: Ubuntu 20.04.2
- Browser/version: Tor-Browser & FF latest stable from Ubuntu repo
- Adblock Extension/version: uBlock Origin 1.37.2

### Other details:

Copy/(moved) issue to https://github.com/uBlockOrigin/uAssets/issues/9968 as suggested by @felix-22

**Comment by felix-22** [comment]
Hey my NFSW buddy, do you mean the `~8000` popit script porn sites I mentioned?: 

> felix-22 : https://github.com/easylist/easylist/issues/8703#issuecomment-903119076 : 
>
> https://publicwww.com/websites/popit/ returns 8151 web pages,
> and "only" 260 are fixed already in uBO: uBlockOrigin/uAssets@7b1f737

**Comment by spirillen** [comment]
> Hey my NFSW buddy, do you mean the `~8000` popit script porn sites I found [#8703 (comment)](https://github.com/easylist/easylist/issues/8703#issuecomment-903119076) ?
> If so, then you most likely want to report them to [uAssets Tracker](https://github.com/uBlockOrigin/uAssets/issues), because they can't be fixed in EasyList...

Hey @felix-22 That's right, I'll move the post a tad later when I've complete the circus...  Right now I''m just keeping the head clear enough to remember where I've been and whats next :unamused: 

But you are most welcome to open a issue at https://github.com/uBlockOrigin/uAssets/issues and the reference this issue :smiley:  

The real secret lies in my commit script :wink:

**Comment by felix-22** [comment]
> Porn Domains: This is an endeavour to find all porn domains and list them in the shortest as possible format it can be done. 

> hosts.txt: `14.243` lines

Hey buddy, have you been aware of: https://github.com/elyoas/hosts/blob/main/shallalist%20(not%20used)/shallalist%20domains%20(20210628) ? It contains over `800.000` lines...

**Comment by spirillen** [comment]
> > Porn Domains: This is an endeavour to find all porn domains and list them in the shortest as possible format it can be done.
> 
> > hosts.txt: `14.243` lines

Not sure which repo you refer to here as there can be several.

## A little timeline history

It all started with https://github.com/Clefspeare13/pornhosts, who at that time started to be more offline than online, therefor I forked the repo to https://github.com/Import-External-Sources/pornhosts and made a lot of scripts to automate the testing and publishing of active lists only based on [PyFunceble](https://github.com/funilrys/PyFunceble) by @funilrys.

Then in the middle of all my testing and programming, @Clefspeare13 trow me the merge keys to https://github.com/Clefspeare13/pornhosts for never to be seen again.

Since I only could merge in his repo, and couldn't actually control it, i set off to establish my own repository [Porn Records](https://mypdns.org/my-privacy-dns/porn-records), which like very all other of my BlackLists should be targeting the [DNS RPZ](https://mypdns.org/mypdns/support/-/wikis/rpz) ([DNS](https://mypdns.org/mypdns/support/-/wikis/dns/)).

All this said I can also acknowledge that the majority is still run on a 10 years outdated hosts file. Then how to understand the "hosts" file? Well you need to do the match in [Combining the lists](https://mypdns.org/my-privacy-dns/porn-records/-/tree/master/#combining-the-lists) to actually makes it spin :wink: 

Another thing you should think of, when it comes tot eh numbers, with RPZ the following three records kills ~500.000 domains with no more than these 3 lines in the DNS

```python
*.adult   CNAME . ; Adult
*.sex   CNAME . ; Adult
*.porn   CNAME . ; Adult
```

If this isn't "shortest as possible" I'm not sure you would expect :smirk: `0.0.0.0 [a-m].root-servers.net` 

.adult: https://mypdns.org/my-privacy-dns/porn-records/-/issues/339
.sex: https://mypdns.org/my-privacy-dns/porn-records/-/issues/660
.porn: https://mypdns.org/my-privacy-dns/porn-records/-/issues/341


> 
> Have you been aware of: https://github.com/elyoas/hosts/blob/main/shallalist%20(not%20used)/shallalist%20domains%20(20210628) ? It contains over `800.000` lines...it's 56x more than your list.

**Comment by felix-22** [comment]
> spirillen : 
> > felix-22 : 
> >
> > Porn Domains: This is an endeavour to find all porn domains and list them in the shortest as possible format it can be done.
> >
> > hosts.txt: `14.243` lines
> 
> Not sure which repo you refer to here as there can be several.

This one: https://github.com/mypdns/porn-records/blob/master/submit_here/adult.mypdns.cloud/hosts.list

> spirillen : Yes and there 699.995 False Positive.... https://github.com/dns-test/porn.host.rpz (Same source, just over DNS)

LOL ~700.000 false positive :smiley: how that even possible?

**Comment by felix-22** [comment]
So as seen, you have led to locking of the two threads:
- https://github.com/uBlockOrigin/uAssets/issues/10003
- https://github.com/uBlockOrigin/uAssets/issues/9968

I hope you won't repeat the same here...

I will repeat it for the last time, the only reason I suggested you to move the issue from EasyList tracker to a proper issue tracker (to uAssets tracker), was just a simple technical fact that the porn sites (scripts) you reported in EasyList tracker can't be fixed in EasyList tracker...it was just a technical hint / suggestion, I was not motivated by any private gain or interest, as you have been trying to accuse me of, and which was the reason you were so unnecessarily "pissed off". If the pages you report are fixed, fine, if they are not, the world will not collapse. As I said, it would be better if you asked and clarified first, rather than unnecessarily misinterpreting and accusing for some imaginary reason. I hope it is now clear who caused all the fuss...You made an accusation, I addressed it, I hope you finally understood it and will not continue to unnecessarily argue or make off-topic. I do not expect a reply, at most "I agree or I apologize", any other answer than that will be addressed not by me but by moderation.
